File: Chapter-04-Strings-AssociativeArrays/Max-Min-Avg-With-Object/max-min-avg-with-object.js
// Given an array of numbers, return an object containing the array's max, min and average values
//  Dont use built-ins

function maxMinAvg(arr){
  if (arr.length){
    var result = {'max': arr[0], 'min': arr[0], 'avg': 0}
    var len = arr.length, sum = arr[0]

    for (let i = 1; i < len; i++){
      result['max'] = ((result['max'] < arr[i]) ? arr[i] : result['max'])
      result['min'] = ((result['min'] > arr[i]) ? arr[i] : result['min'])
      sum += arr[i]
    }

    result['avg'] = sum/len

    return result
  } else {
    return "Empty Array"
  }
}

var myArr = [1,2,3,4,5]
myArr = [-1,-2,3,-4,5,0,0]
myArr = []
myArr = [1]
console.log(maxMinAvg(myArr))

File: Chapter-04-Strings-AssociativeArrays/Max-Min-Avg-With-Object/max-min-avg-with-object.py
# Given an array of numbers, return an object containing the array's max, min and average values
# Dont use built-ins

def maxMinAvg(arr):
    if len(arr):
        result = {'max': arr[0], 'min': arr[0], 'avg': 0}
        l = len(arr)
        s = arr[0]

        for i in range(1, l):
            result['max'] = arr[i] if result['max'] < arr[i] else result['max']
            result['min'] = arr[i] if result['min'] > arr[i] else result['min']
            s += arr[i]

        result['avg'] = s / (l * 1.0)

        return result
    else:
        return "Empty String"

myArr = [1,2,3,4,5]
myArr = [-1,-2,3,-4,5,0,0]
myArr = []
myArr = [1]
print(maxMinAvg(myArr))
